plat/linuxu/include/linuxu/syscall.h

```diff
@@ -123,8 +129,12 @@ static inline int sys_open(const char *pathname, int flags, ...)
 		mode = va_arg(ap, mode_t);
 		va_end(ap);
 	}
-
+#if defined __ARM_64__
+	fd = syscall4(__SC_OPENAT, AT_FDCWD, (long)pathname, (long)flags,
```

Comment:
I think it would simplify our code if we get x86 and Arm running with the same system call. I learned that on Arm there are less syscalls available but they are usually the most powerful variant. On x86 `OPEN` but also `OPENAT` are available. `OPENAT` can be considered as the more powerful variant as `OPEN`. Do you think we can let x86 and Arm use both `OPENAT`? This would make the architecture-dependent branching unnecessary here.

Reply:
Yes, I think it will be okay. That means that I will have to add a new define in each header file (for x86_64 and arm) for the `OPENAT` system call.

Reply:
:+1: correct